lib/src/default_index/rev_walk.rs

```diff
@@ -643,6 +668,16 @@ impl RevWalk<CompositeIndex> for RevWalkDescendantsImpl {
     }
 }
 
+/// Filter a slice by a range, clamping the range to the length of the slice.
+fn filter_slice_by_range<'a, T: Copy>(
+    slice: &'a [T],
+    range: &Range<u32>,
+) -> impl Iterator<Item = T> + use<'a, T> {
+    let start = (range.start as usize).min(slice.len());
+    let end = (range.end as usize).min(slice.len());
+    slice[start..end].iter().copied()
```

Comment:
nit: we'll still need to handle `start > end` somewhere, maybe in `RevWalkBuilder`.